src/linux/component.rs

```diff
@@ -258,25 +261,37 @@ fn append_files(components: &mut Vec<Component>, folder: &Path) -> Option<()> {
         component.name = name.unwrap_or(String::new());
         let device_model = get_file_line(&folder.join("device/model"), 16);
         component.device = device_model.unwrap_or(String::new());
-        fill_component(component, id, item, filename);
+        fill_component(component, id, item, folder, filename);
     }
-    let compo = matchings.into_iter().map(|(k, mut c)| {
-        // sysinfo expose a generic interface with a `label`.
-        // Problem: lot of sensor don't have a label or a device model! ¯\_(ツ)_/¯
-        // So let's pretend we have a unique label!
-        //
-        // All sensor have a `name` on linux hwmon it's mandatory.
-        // - if a sensor has no label: check if don't have a model name, if so it's use it.
-        // - if not do something from id and name at best.
-        if c.label.is_empty() {
-            c.label = if c.device.is_empty() {
-                format!("{} {}", c.name, k)
-            } else {
-                format!("{} {} {}", c.name, c.device, k)
-            }
-        }
-        c
-    });
+    let compo = matchings
+        .into_iter()
+        .map(|(k, mut c)| {
+            // sysinfo expose a generic interface with a `label`.
+            // Problem: lot of sensor don't have a label or a device model! ¯\_(ツ)_/¯
+            // So let's pretend we have a unique label!
+            //
+            // All sensor have a `name` on linux hwmon it's mandatory.
+            // - if a sensor has no label: check if don't have a model name, if so it's use it.
+            // - if not do something from id and name at best.
+            let has_device = !c.device.is_empty();
+            let has_label = !c.label.is_empty();
+            c.label = match (has_label, has_device) {
+                (true, true) => format!("{} {} {}", c.name, c.label, c.device),
+                (true, false) => format!("{} {}", c.name, c.label),
+                (false, true) => format!("{} {}", c.name, c.device),
+                (false, false) => format!("{} {}", c.name, k),
+            };
+            c
+        })
+        // Remove component without `tempN_input file termal Component don't support this kind of sensors.
+        .filter(|c| {
+            !c.input_file
+                .as_path()
+                .file_name()
+                .and_then(|f| f.to_str())
+                .unwrap_or("")
+                .is_empty()
```

Comment:
Little simplification (because `OsStr` has `is_empty`):

```suggestion
                .and_then(|f| f.is_empty())
                .unwrap_or(true)
```